classes.py

```diff
@@ -0,0 +1,544 @@
+import json
+import math
+import os
+from urllib.parse import urlparse, parse_qs
+import csv
+
+import requests
+from bs4 import BeautifulSoup
+
+_CSV_DELIMITER = ';'
+_URLDOMAIN = "https://chess-results.com"
+_RATING_LIST_HEADER = 'Id_No;Id_CBX;Title;Name;Rtg_Nat;ClubName;Birthday;Sex;Fed;TotalNumGames;SumOpponRating' \
+                      ';TotalPoints'
+_MAX_NUM_GAMES_TEMP_RATING = 15
+_K_STARTING_NUM_GAMES = [(30, 0),  # grampo
+                         (25, _MAX_NUM_GAMES_TEMP_RATING),  # 15
+                         (15, 40),
+                         (10, 80)]
+_CURRENT_TOURNAMENT_SNR = 0
+
+
+class FexerjRatingCycle:
+
+    def __init__(self, tournaments_file, first_item, items_to_process, initial_rating_filepath):
+
+        self.tournaments_file = tournaments_file
+        self.first_item = first_item
+        self.items_to_process = items_to_process
+        self.rating_list = {}
+        self.tournaments = {}
+        self.cbx_to_fexerj = {}
+        self.manual_entries = {}
+
+        self.load_manual_entry_dict()
+
+        with open(tournaments_file, 'r') as f:
+            reader = csv.reader(f, delimiter=_CSV_DELIMITER)
+            self.tournaments = list(reader)[1:]
+            for tournament in self.tournaments:
+
+                final_rating_filepath = "RatingList_after_%s.csv" % (tournament[0])
+
+                if int(tournament[0]) in range(first_item, first_item + items_to_process):
+
+                    print("\nRunning tournament %s (%s)...\n" % (tournament[0], tournament[2]))
+
+                    self.get_rating_list(initial_rating_filepath)
+
+                    print("Reading from %s" % initial_rating_filepath)
+                    print("Writing to %s" % final_rating_filepath)
+
+                    trn_type = tournament[4]
+
+                    if trn_type == 'SS':
+                        tournament = SwissSingleTournament(self, tournament)
+                    elif trn_type == 'RR':
+                        tournament = RoundRobinTournament(self, tournament)
+                    elif trn_type == 'ST':
+                        tournament = SwissTeamTournament(self, tournament)
+                    else:
+                        raise ValueError('Wrong tournament type: %s' % trn_type)
+
+                    tournament.complete_players_info()
+                    tournament.calculate_players_ratings()
+                    tournament.write_new_ratings_list(final_rating_filepath)
+
+                initial_rating_filepath = final_rating_filepath
+
+        self.write_manual_entry_dict()
+
+    def get_rating_list(self, initial_rating_filepath):
+
+        with open(initial_rating_filepath) as rating_list:
+            reader = csv.reader(rating_list, delimiter=_CSV_DELIMITER)
+            next(reader, None)  # Skip the headers
+            for row in reader:
+                self.rating_list.update({int(row[0]): FexerjPlayer(int(row[0]),  # ID FEXERJ
```

Comment:
Not sure if this is just in my screen, but could we reformat this in a way that the list stays a bit more to the left of the screen? For me they all show up as wrapped lines.

Reply:
The method call has many parameters so a new line was added after each parameter to improve readability.

![image](https://github.com/felipeamp/fexerj-rating-calculator/assets/29162354/16277f10-bf52-442f-b908-da59eb7a6e9b)

Could you please explain better how exactly do you want this to be presented?

Reply:
yeah, this is not great, but I'm not sure of a great way of doing it. Keyword arguments would have pretty much the same issue. Maybe if you extracted the part that constructs the `FexerjPlayer` into a local variable outside of the method call the indentation would be smaller. Something like this:

```suggestion
for row in reader:
    player = FexerjPlayer(int(row[0]),  # ID FEXERJ
                                           row[1],  # ID CBX
                                           row[2],  # TITLE
                                           row[3],  # NAME
                                           int(row[4]),  # RATING
                                           row[5],  # CLUB
                                           row[6],  # BIRTHDAY
                                           row[7],  # SEX
                                           row[8],  # FEDERATION
                                           int(row[9]),  # TOTAL NUM OF GAMES
                                           row[10],  # SUM OF OPPONENT RATINGS
                                           row[11])  # POINTS AGAINST OPPONENTS
    self.rating_list.update({int(row[0]): player})
```

Reply:
Fixed